homeassistant/components/smarty/fan.py

```diff
@@ -0,0 +1,133 @@
+"""
+Platform to control a Salda Smarty XP/XV ventilation unit.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/fan.smarty/
+"""
+import logging
+
+from homeassistant.core import callback
+from homeassistant.components.fan import (
+    SPEED_HIGH, SPEED_LOW, SPEED_MEDIUM, SPEED_OFF, SUPPORT_SET_SPEED,
+    FanEntity)
+from homeassistant.components.smarty import (
+    DATA_SMARTY, SIGNAL_UPDATE_SMARTY, SMARTY_NAME)
+from homeassistant.const import (STATE_OFF, STATE_ON)
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+DEPENDENCIES = ['smarty']
+
+_LOGGER = logging.getLogger(__name__)
+
+SPEED_LIST = [SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
+
+SPEED_MAPPING = {
+    1: SPEED_LOW,
+    2: SPEED_MEDIUM,
+    3: SPEED_HIGH
+}
+
+SPEED_TO_MODE = {
+    SPEED_LOW: 1,
+    SPEED_MEDIUM: 2,
+    SPEED_HIGH: 3
+}
+
+
+async def async_setup_platform(hass, config,
+                               async_add_devices, discovery_info=None):
+    """Set up the Smarty Fan Platform."""
+    smarty = hass.data[DATA_SMARTY]
+    name = hass.data[SMARTY_NAME]
+
+    async_add_devices([SmartyFan(name, smarty)])
+
+
+class SmartyFan(FanEntity):
+    """Representation of a Smarty Fan."""
+
+    def __init__(self, name, smarty):
+        """Initialize the entity."""
+        self._name = name
+        self._speed = SPEED_OFF
+        self._state = None
+        self._smarty = smarty
+
+    @property
+    def should_poll(self):
+        """Do not poll."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the fan."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return 'mdi:air-conditioner'
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_SET_SPEED
+
+    @property
+    def speed_list(self):
+        """List of available fan modes."""
+        return SPEED_LIST
+
+    @property
+    def is_on(self):
+        """Return state of the fan."""
+        return self._state == STATE_ON
+
+    @property
+    def speed(self) -> str:
+        """Return speed of the fan."""
+        return self._speed
+
+    def turn_on(self, speed=None, **kwargs):
+        """Turn on the fan."""
+        _LOGGER.debug('Turning on fan. Speed is %s', speed)
+        if speed is None:
+            self._speed = SPEED_MEDIUM
```

Comment:
Shouldn't we check if the call is successful before assigning the new speed, like we do below?

Reply:
yes, thanks.